core/src/main/java/hudson/model/StringParameterDefinition.java

```diff
@@ -145,7 +145,7 @@ public String getHelpFile() {
     @Override
     public ParameterValue createValue(StaplerRequest req, JSONObject jo) {
         StringParameterValue value = req.bindJSON(StringParameterValue.class, jo);
-        if (isTrim() && value!=null) {
+        if (isTrim()) {
```

Comment:
As of #5275

https://github.com/jenkinsci/jenkins/blob/165d559469c7a58af581931bd1b89e5b9ed4a9af/core/src/main/java/hudson/model/StringParameterValue.java#L51

means that this can no longer be null.